Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/repo/DefaultContactDiaryRepository.kt

```diff
@@ -0,0 +1,205 @@
+@file:Suppress("TooManyFunctions")
+package de.rki.coronawarnapp.contactdiary.storage.repo
+
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPerson
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPersonEncounter
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryPersonEncounter
+import de.rki.coronawarnapp.contactdiary.model.sortByNameAndIdASC
+import de.rki.coronawarnapp.contactdiary.storage.dao.ContactDiaryLocationDao
+import de.rki.coronawarnapp.contactdiary.storage.dao.ContactDiaryLocationVisitDao
+import de.rki.coronawarnapp.contactdiary.storage.dao.ContactDiaryPersonDao
+import de.rki.coronawarnapp.contactdiary.storage.dao.ContactDiaryPersonEncounterDao
+import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryLocationVisitEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryPersonEncounterEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.toContactDiaryLocationEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.toContactDiaryLocationVisitEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.toContactDiaryPersonEncounterEntity
+import de.rki.coronawarnapp.contactdiary.storage.entity.toContactDiaryPersonEntity
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.joda.time.LocalDate
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DefaultContactDiaryRepository @Inject constructor(
+    private val contactDiaryLocationDao: ContactDiaryLocationDao,
+    private val contactDiaryLocationVisitDao: ContactDiaryLocationVisitDao,
+    private val contactDiaryPersonDao: ContactDiaryPersonDao,
+    private val contactDiaryPersonEncounterDao: ContactDiaryPersonEncounterDao
+) : ContactDiaryRepository {
+
+    // Location
+    override val locations: Flow<List<ContactDiaryLocation>> = contactDiaryLocationDao
+        .allEntries()
+        .map { it.sortByNameAndIdASC() }
+
+    override suspend fun addLocation(contactDiaryLocation: ContactDiaryLocation) {
+        Timber.d("Adding location $contactDiaryLocation")
```

Comment:
Shouldn't the string here use placeholders and pass `contactDiaryLocation` as second parameter to `d()` for efficiency and to avoid unnecessary `toString` invocations?

Reply:
I'm not sure about this...